falcon/hooks.py

```diff
@@ -15,13 +15,28 @@
 """Hook decorators."""
 
 from functools import wraps
+import inspect
 
 import six
 
 from falcon import COMBINED_METHODS
 from falcon.util.misc import get_argnames
 
 
+def predicate_method_function(member):
```

Comment:
I am thinking to revert this approach by @mumrau in the follow-up PR. While I personally like it being very clean, nice and explicit, but I think just listing all the members, and checking those that quack a *callable*, are probably good enough to decorate.

Consider the following example:

```python
import inspect

import falcon


def predicate_method_function(member):
    return inspect.ismethod(member) or inspect.isfunction(member)


class HelloSayer(object):

    def __call__(self, req, resp):
        resp.media = {'Hello': 'World'}


class Resource(object):

    on_get = HelloSayer()


api = falcon.API()
api.add_route('/hello', Resource())

print(inspect.getmembers(Resource, predicate_method_function))
```

It is undeniably a bit contrived, but OTOH we cannot know how exactly people are building their APIs.

It does say Hello, but ``inspect.getmembers`` returns an empty list with the exemplified predicate function.

Reply:
Makes sense. Totally didn't think of that approach, stayed focused on the "on_http(_optional_follow_up_subhandler)".
I agree with the callables inspections then.